src/app/report/report.component.ts

```diff
@@ -1,64 +1,70 @@
-
-@Injectable()
-export class ReportData {
-  data: {} = {};
-  defaultDisplay: any;
-  currentView: any;
-}
+import { TabService } from '../shared/services/tab.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { View } from '../shared/interfaces/view';
+import { ReportData } from '../shared/interfaces/report-data';
+import { DebugComponent } from '../debug/debug.component';
 
 @Component({
   selector: 'app-report',
   templateUrl: './report.component.html',
   styleUrls: ['./report.component.css'],
 })
-export class ReportComponent implements AfterViewInit {
+export class ReportComponent implements OnInit, AfterViewInit {
+  static readonly ROUTER_PATH: string = 'report';
   @ViewChild('treeReference') treeReference!: jqxTreeComponent;
   @ViewChild(EditDisplayComponent) editDisplayComponent!: EditDisplayComponent;
   @ViewChild(DisplayComponent) displayComponent!: DisplayComponent;
-  defaultDisplay: boolean = false;
-  currentView: any = {};
+  currentView?: View;
+  reportData?: ReportData;
 
   constructor(
-    public reportData: ReportData,
+    private tabService: TabService,
     private helperService: HelperService,
     private dynamicService: DynamicService,
+    private route: ActivatedRoute,
+    private router: Router,
   ) {}
 
+  ngOnInit(): void {
+    this.reportData = this.tabService.activeReportTabs.get(this.getIdFromPath());
+    if (!this.reportData) {
+      this.router.navigate([DebugComponent.ROUTER_PATH]);
+    }
+  }
+
   ngAfterViewInit(): void {
     setTimeout(() => {
-      this.createTree(<Report>(<unknown>this.reportData));
-      this.treeReference.selectItem(this.treeReference.getItems()[0]);
-      this.currentView = this.reportData.currentView;
-      if (this.reportData.defaultDisplay) {
-        this.displayComponent?.showReport(this.reportData);
-      } else {
-        this.editDisplayComponent?.showReport(this.reportData);
+      if (this.reportData) {
+        this.createTree(this.reportData.report);
+        this.treeReference.selectItem(this.treeReference.getItems()[0]);
+        this.currentView = this.reportData.currentView;
+        this.editDisplayComponent.showReport(this.reportData.report);
       }
     });
   }
 
-  createTree(report: Report) {
+  getIdFromPath(): string {
+    return this.route.snapshot.paramMap.get('id') as string;
+  }
+
+  createTree(report: Report): void {
     let tree = this.helperService.convertReportToJqxTree(report);
     this.treeReference.createComponent({ height: '100%', width: '100%', source: [tree], allowDrag: false });
   }
 
   selectReport(currentReport: any): void {
     let report = currentReport.owner.selectedItem.value;
-    if (this.reportData.defaultDisplay) {
-      this.displayComponent.showReport(report);
-    } else {
-      this.editDisplayComponent.showReport(report);
-    }
+    this.editDisplayComponent.showReport(report);
   }
 
-  savingReport(report: any) {
+  savingReport(report: any): void {
```

Comment:
Is this report not of the same type report?

Reply:
Fixed